core/src/main/java/org/elasticsearch/index/shard/SearchOperationListener.java

```diff
@@ -225,5 +233,25 @@ public void onFreeScrollContext(SearchContext context) {
                 }
             }
         }
+
+        @Override
+        public void validateSearchContext(SearchContext context) throws Exception {
+            Exception exception = null;
+            for (SearchOperationListener listener : listeners) {
+                try {
+                    listener.validateSearchContext(context);
+                } catch (Exception e) {
+                    if (exception == null) {
```

Comment:
you can use `exception = ExceptionsHelper#useOrSuppress(exception, e)` here instead